s3/src/main/scala/geotrellis/spark/io/s3/S3GeoTiffInfoReader.scala

```diff
@@ -0,0 +1,190 @@
+package geotrellis.spark.io.s3
+
+import geotrellis.raster.io.geotiff.reader.GeoTiffReader
+import geotrellis.spark.io.RasterReader
+import geotrellis.spark.io.s3.util.S3RangeReader
+import geotrellis.raster.io.geotiff.reader.GeoTiffReader.GeoTiffInfo
+import geotrellis.util.LazyLogging
+
+import com.amazonaws.services.s3.model.ListObjectsRequest
+import org.apache.spark.SparkContext
+import org.apache.spark.rdd.RDD
+
+import scala.collection.mutable
+
+case class S3GeoTiffInfoReader(
```

Comment:
By the way this is used it looks like it could be decomposed into a companion object with some methods on it, that would probably be safer than a case class that potentially carries client around.